### PR TITLE
Subscription Mode & Enhanced deployment architecture

### PR DESCRIPTION
**Pull request by dusanstanis**
Base for subscription/bonus logic.

### CODE REVIEW

File: test-producer.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+const { Producer } = require("sqs-producer");
```
Comment (rekpero) [inline test-producer.js:1]: @dusanstanis remove this file, and put it in the gitignore

File: test-consumer.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const { Consumer } = require("sqs-consumer");
```
Comment (rekpero) [inline test-consumer.js:1]: @dusanstanis remove this file, and put it in the gitignore

File: src/components/Subscription/model.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import * as connections from "../../config/connection/connection";
+import { Document, Schema } from "mongoose";
+import { IOrganization } from "../Organization/model";
+import { ISubscriptionPackage } from "../SubscriptionPackage/model";
+
+export interface ISubscription extends Document {
```
Comment (rekpero) [inline src/components/Subscription/model.ts:6]: put this interface in `interfaces.ts`

File: src/components/SharedParameter/model.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import * as connections from "../../config/connection/connection";
+import { Document, Schema } from "mongoose";
+
+export interface ISharedParameter extends Document {
```
Comment (rekpero) [inline src/components/SharedParameter/model.ts:4]: move this to interfaces.ts file

File: src/components/SharedParameter/service.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import SharedParameterModel, { ISharedParameter } from "./model";
+
+const SharedParameterService = {
```
Comment (rekpero) [inline src/components/SharedParameter/service.ts:3]: create an interface for this service and put it in the interfaces.ts

File: src/components/Subscription/service.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import { ISubscriptionPackage } from "../SubscriptionPackage/model";
+import SubscriptioneModel, { ISubscription } from "./model";
+
+const SubscriptionService = {
```
Comment (rekpero) [inline src/components/Subscription/service.ts:4]: create an interface for service and put it in interfaces.ts file

File: src/components/Bonus/service.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import BonusModel, { IBonus } from "./model";
+import { ISharedParameter } from "../SharedParameter/model";
+import SharedParameterService from "../SharedParameter/service";
+
+const BonusService = {
```
Comment (rekpero) [inline src/components/Bonus/service.ts:5]: create an interface for this service and put it in interfaces.ts

File: src/components/Bonus/model.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import * as connections from "../../config/connection/connection";
+import { Document, Schema } from "mongoose";
+import { ISubscription } from "../Subscription/model";
+
+export interface IBonus extends Document {
```
Comment (rekpero) [inline src/components/Bonus/model.ts:5]: put this in interfaces.ts

File: src/components/SubscriptionPackage/model.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
+import * as connections from "../../config/connection/connection";
+import { Document, Schema } from "mongoose";
+
+export interface ISubscriptionPackage extends Document {
```
Comment (rekpero) [inline src/components/SubscriptionPackage/model.ts:4]: put this in interfaces.ts file

File: src/components/SubscriptionPackage/service.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
+import SubscriptionPackageModel, { ISubscriptionPackage } from "./model";
+import SharedParameterService from "../SharedParameter/service";
+import { ISharedParameter } from "../SharedParameter/model";
+
+const SubscriptionPackageService = {
```
Comment (rekpero) [inline src/components/SubscriptionPackage/service.ts:5]: create an interface for this service and put it in interfaces.ts file

**Review by rekpero — CHANGES_REQUESTED** [review]
@dusanstanis code looks good, just a little bit of refactoring is needed on the interface side but good to merge.

**Comment by rekpero** [comment]
Also, @dusanstanis do run `num run prettier:write` before commiting

File: src/components/Deployment/dto-interfaces.ts [hunk 11]
```diff
@@ -32,6 +32,7 @@ export interface IDeploymentBody {
   walletId: string;
   walletAddress: string;
   env: any;
+  payedViaSubscription: boolean;
```
Comment (akinovak) [inline src/components/Deployment/dto-interfaces.ts:35]: paid?

**Review by rekpero — APPROVED** [review]

**Pull request merged**